bats/sql-reset.bats

```diff
@@ -172,3 +172,80 @@ teardown() {
      [[ "$output" =~ pk ]] || false
      [[ "$output" != 1  ]] || false
 }
+
+@test "DOLT_RESET --hard properly wipes diff system table immediately" {
```

Comment:
Wouldn't phrase this as wiping a system table

Maybe "no rows in dolt_diff table after reset --hard"